src/AirtableClient.php

```diff
@@ -45,58 +68,91 @@ public function findAll(string $table, ?string $view = null): array
      * @param  mixed $table Table name
      * @param  mixed $field Search field name
      * @param  mixed $value Wanted value
+     * @param  string $dataClass The class name which will hold fields data
      * @return array
      */
-    public function findBy(string $table, string $field, string $value): array
+    public function findBy(string $table, string $field, string $value, ?string $dataClass = null): array
     {
-        $filterByFormula = "?filterByFormula=AND({".$field."} = '".$value."')";
-        $url = $this->id .'/'. $table . $filterByFormula;
+        $filterByFormula = "?filterByFormula=AND({" . $field . "} = '" . $value . "')";
+        $url = $this->id . '/' . $table . $filterByFormula;
         $response = $this->request($url);
 
-        return $response->toArray()['records'];
+        $airtableRecords = array_map(
+            fn ($record) => AirtableRecord::fromRecord($record),
+            $response->toArray()['records']
+        );
+
+        if ($dataClass) {
+            foreach ($airtableRecords as $record) {
+                $record->setFields(
```

Comment:
IMO this setter should be remove and the tranformed fields should be pass to the method fromRecord  there is some duplication code, you should refacto the transformation logic, I mean the logic to tranform a row to a AirtableLogic should be done in one method and you call it in a loop.

Reply:
The logic has been fixed, I left the setter but it could be removed if we think it is not necessary anymore